### PR TITLE
Update to be compatible with styhead and walnascar

### PR DESCRIPTION
**Pull request by xbrtszstk**
Here you have necessary changes to work with latest Yocto releases.
There are still some issues to resolve:
1. I updated the scarthgap branch, but it should be merged with styhead or master. I choose this one, because styhead does not exist and master is far behind scarthgap.
2. the Yocto requires Upstream-Status in each patch. I marked it as Inappropriate, but you should choose correct one.
3. rive-taffy-ffi returns "Did not find anything to install" in the do_install function.
4. In flutter-auto and ivi-homescreen I changed the destsuffix in SRC_URI, because it was causing incompatible source directory structure. ivi-homescreen sources were in ${WORKDIR}/git/git and ivi-homescreen-plugins sources were in ${WORKDIR}/git/ivi-homescreen-plugins. In this case yocto fails, because the LICESNE, and CMakeLists.txt files were in wrong directory. Currently, ivi-homescreen sources are in ${WORKDIR}/git and ivi-homescreen-plugins sources are in ${UNPACKDIR}/ivi-homescreen-plugins. Recipes build without issue but I'm not sure if they can find the plugins directory.
5. many of examples from meta-flutter-apps have issues with buildpaths.

I noticed also other issues:
1. the libwebrtc recipe requires multimedia-layer, which is not required layer, so it is unbuildable with default configuration.
2. the packagegroup-flutter-sdk-deps recipe requires clang-layer, which is not required layer, so it is unbuildable with default configuration.

### CODE REVIEW

File: dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0001-gn-riscv32-and-riscv64.patch [hunk 1]
```diff
@@ -1,4 +1,5 @@
 From 748e416e394552cac6e4be30112c3ef45d1c0ec3 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0001-gn-riscv32-and-riscv64.patch:2]: since this is agnostic to branch, should be in separate PR

File: dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0002-fml-build-config-add-riscv.patch [hunk 2]
```diff
@@ -1,4 +1,5 @@
 From 8b34fbdace418dcf338492ac63ce1048ecc20bf5 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0002-fml-build-config-add-riscv.patch:2]: see above - separate PR

File: dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0003-swiftshader-riscv-support.patch [hunk 3]
```diff
@@ -1,4 +1,5 @@
 From 9d3788480dba3dcc27cd00d4852c968b0e5de502 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0003-swiftshader-riscv-support.patch:2]: see above - separate PR

File: dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0004-tonic-riscv-support.patch [hunk 4]
```diff
@@ -1,4 +1,5 @@
 From 9be5e09115a693ab7fc8cee2122d0a4318fc6e71 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline dynamic-layers/clang-layer/recipes-graphics/flutter-engine/files/0004-tonic-riscv-support.patch:2]: see above - separate PR

File: meta-flutter-apps/recipes-graphics/flutter-apps/first-party/files/super_dash/0001-intl-0.19.0.patch [hunk 5]
```diff
@@ -1,4 +1,5 @@
 From 273a62a435e184499cabcccbeb5c50a99735f8e2 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline meta-flutter-apps/recipes-graphics/flutter-apps/first-party/files/super_dash/0001-intl-0.19.0.patch:2]: see above - separate PR

File: meta-flutter-apps/recipes-graphics/flutter-apps/first-party/files/super_dash/0001-ivi-homescreen-support.patch [hunk 6]
```diff
@@ -1,4 +1,5 @@
 From 4977717a5bf7478e894e07005ce04e93953df0bf Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline meta-flutter-apps/recipes-graphics/flutter-apps/first-party/files/super_dash/0001-ivi-homescreen-support.patch:2]: see above - separate PR

File: meta-flutter-apps/recipes-graphics/flutter-apps/third-party/files/sourcya-playx-3d-scene/0001-ivi-homescreen-support.patch [hunk 7]
```diff
@@ -1,4 +1,5 @@
 From 5c3a2fed9b4d2e4b5013c1ba1b40f3d8beccafd9 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline meta-flutter-apps/recipes-graphics/flutter-apps/third-party/files/sourcya-playx-3d-scene/0001-ivi-homescreen-support.patch:2]: see above - separate PR

File: recipes-devtools/sentry/files/0001-version-SO.patch [hunk 8]
```diff
@@ -1,4 +1,5 @@
 From fb83605e554b78f199cccdba42cf58c873b5dd1c Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-devtools/sentry/files/0001-version-SO.patch:2]: see above - separate PR

File: recipes-devtools/sentry/files/0002-RISC-V-support.patch [hunk 9]
```diff
@@ -1,4 +1,5 @@
 From a47ef37dcf1830cc71f1144a87f67a0cd5d8b8a3 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-devtools/sentry/files/0002-RISC-V-support.patch:2]: see above - separate PR

File: recipes-graphics/flutter-engine/files/0001-export-GPU-symbols.patch [hunk 10]
```diff
@@ -1,4 +1,5 @@
 From bfdda9a112f91117c4a13d89a88c27f4d1077dea Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/files/0001-export-GPU-symbols.patch:2]: see above - separate PR

File: recipes-graphics/flutter-engine/files/0002-libcxx-uglify-support-musl.patch [hunk 11]
```diff
@@ -1,4 +1,5 @@
 From 8e19a2b43573c0245c8925d38140d76051dfbed0 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/files/0002-libcxx-uglify-support-musl.patch:2]: see above - separate PR

File: recipes-graphics/flutter-engine/files/0003-libcxx-return-type-in-wcstoull_l.patch [hunk 12]
```diff
@@ -1,4 +1,5 @@
 From 939f5a33711370697f9ad1de4267cfc7399dfe86 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/files/0003-libcxx-return-type-in-wcstoull_l.patch:2]: see above - separate PR

File: recipes-graphics/flutter-engine/files/0004-suppres-musl-libc-warning.patch [hunk 13]
```diff
@@ -1,4 +1,5 @@
 From 78672cfb355bc4dae7f6729b52858798a11a869d Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/files/0004-suppres-musl-libc-warning.patch:2]: see above - separate PR

File: recipes-devtools/dart/dart-sdk_3.5.4.bb [hunk 14]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020-2023 Joel Winarske. All rights reserved.
+# Copyright (c) 2020-2024 Joel Winarske. All rights reserved.
```
Comment (jwinarske) [inline recipes-devtools/dart/dart-sdk_3.5.4.bb:2]: see above - separate PR

File: recipes-graphics/flutter-engine/flutter-engine_git.bb [hunk 15]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020-2023 Joel Winarske. All rights reserved.
+# Copyright (c) 2020-2024 Joel Winarske. All rights reserved.
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/flutter-engine_git.bb:2]: see above - separate PR

File: recipes-graphics/flutter-engine/flutter-engine_git.bb [hunk 16]
```diff
@@ -106,7 +106,7 @@ PACKAGECONFIG[impeller-3d] = "--enable-impeller-3d"
 
 CLANG_BUILD_ARCH = "${@clang_build_arch(d)}"
 CLANG_TOOLCHAIN_TRIPLE = "${@gn_clang_triple_prefix(d)}"
-CLANG_PATH = "${WORKDIR}/src/flutter/buildtools/linux-${CLANG_BUILD_ARCH}/clang"
+CLANG_PATH = "${S}/flutter/buildtools/linux-${CLANG_BUILD_ARCH}/clang"
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/flutter-engine_git.bb:109]: leave as is, so PR is specific to `master`

File: recipes-graphics/flutter-engine/flutter-engine_git.bb [hunk 17]
```diff
@@ -209,10 +209,10 @@ do_configure() {
 
         # make it easy to parse
         BUILD_DIR="$(echo ${TMP_OUT_DIR} | sed "s/_RUNTIME_/${MODE}/g")"
-        ARGS_FILE="${WORKDIR}/src/${BUILD_DIR}/args.gn"
+        ARGS_FILE="${S}/${BUILD_DIR}/args.gn"
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/flutter-engine_git.bb:212]: leave as is, so PR is specific to changes required for `master`.  This is a style change

File: recipes-graphics/flutter-engine/flutter-engine_git.bb [hunk 18]
```diff
@@ -209,10 +209,10 @@ do_configure() {
...
 
         # remove in case this is a rebuild and you're not using rm_work.bbclass
-        rm -rf ${WORKDIR}/src/${BUILD_DIR} | true
+        rm -rf ${S}/${BUILD_DIR} | true
```
Comment (jwinarske) [inline recipes-graphics/flutter-engine/flutter-engine_git.bb:215]: leave as is, so PR is specific to changes required for `master`.  This is a style change

File: recipes-graphics/flutter-sdk/flutter-sdk_git.bb [hunk 19]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020-2023 Joel Winarske. All rights reserved.
+# Copyright (c) 2020-2024 Joel Winarske. All rights reserved.
```
Comment (jwinarske) [inline recipes-graphics/flutter-sdk/flutter-sdk_git.bb:2]: not related to changes required for master, separate PR

File: recipes-graphics/libwebrtc/files/add-libwebrtc-target.patch [hunk 20]
```diff
@@ -1,4 +1,5 @@
 From fb0f7a0e7ab3d5965126e2572fe1e219187e27ee Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/libwebrtc/files/add-libwebrtc-target.patch:2]: not related to changes required for master, separate PR

File: recipes-graphics/pdfium/files/public_headers.patch [hunk 21]
```diff
@@ -1,3 +1,4 @@
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/pdfium/files/public_headers.patch:1]: not related to changes required for master, separate PR

File: recipes-graphics/pdfium/files/shared_library.patch [hunk 22]
```diff
@@ -1,3 +1,4 @@
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/pdfium/files/shared_library.patch:1]: not related to changes required for master, separate PR

File: recipes-graphics/pdfium/files/v8_init.patch [hunk 23]
```diff
@@ -1,3 +1,4 @@
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/pdfium/files/v8_init.patch:1]: not related to changes required for master, separate PR

File: recipes-graphics/sony/files/0001-Add-missing-stdint-header.patch [hunk 24]
```diff
@@ -1,4 +1,5 @@
 From d55ded3a6705cb7a834cbe0250aeabf9d82a24c1 Mon Sep 17 00:00:00 2001
+Upstream-Status: Inappropriate
```
Comment (jwinarske) [inline recipes-graphics/sony/files/0001-Add-missing-stdint-header.patch:2]: not related to changes required for master, separate PR

File: recipes-graphics/toyota/flutter-auto_2.0.bb [hunk 25]
```diff
@@ -27,7 +27,7 @@ PLUGINS_COMMIT ??= "bbcf71a9e90e55ca4b63229b9be4e6ba995eb241"
 
 SRC_URI = "\
     gitsm://github.com/toyota-connected/ivi-homescreen.git;protocol=https;branch=v2.0;name=homescreen \
-    git://github.com/toyota-connected/ivi-homescreen-plugins.git;protocol=https;branch=v2.0;name=plugins;destsuffix=${S}/ivi-homescreen-plugins \
+    git://github.com/toyota-connected/ivi-homescreen-plugins.git;protocol=https;branch=v2.0;name=plugins;destsuffix=ivi-homescreen-plugins \
```
Comment (jwinarske) [inline recipes-graphics/toyota/flutter-auto_2.0.bb:30]: not related to changes required for master, separate PR

File: recipes-graphics/toyota/ivi-homescreen_2.0.bb [hunk 26]
```diff
@@ -27,7 +27,7 @@ PLUGINS_COMMIT ??= "bbcf71a9e90e55ca4b63229b9be4e6ba995eb241"
 
 SRC_URI = "\
     gitsm://github.com/toyota-connected/ivi-homescreen.git;protocol=https;branch=v2.0;name=homescreen \
-    git://github.com/toyota-connected/ivi-homescreen-plugins.git;protocol=https;branch=v2.0;name=plugins;destsuffix=${S}/ivi-homescreen-plugins \
+    git://github.com/toyota-connected/ivi-homescreen-plugins.git;protocol=https;branch=v2.0;name=plugins;destsuffix=ivi-homescreen-plugins \
```
Comment (jwinarske) [inline recipes-graphics/toyota/ivi-homescreen_2.0.bb:30]: not related to changes required for master, separate PR.  This is a style change

File: recipes-platform/images/app-container-user.bb [hunk 27]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020-2023 Joel Winarske. All rights reserved.
+# Copyright (c) 2020-2024 Joel Winarske. All rights reserved.
```
Comment (jwinarske) [inline recipes-platform/images/app-container-user.bb:2]: not related to changes required for master, separate PR

**Review by jwinarske — CHANGES_REQUESTED** [review]
The general idea is the minimum viable PR to support changes required for `master` to build.

Agnostic changes that apply to master, scarthgap, kirkstone, and dunfell should be separate PR.  This allows use of `git cherry-pick` on all the supported branches.

File: recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb [hunk 28]
```diff
@@ -2,7 +2,7 @@
 # Copyright (c) 2024 Joel Winarske. All rights reserved.
 #
 
-SUMMARY = "libbrtc"
+SUMMARY = "libwebrtc"
```
Comment (xbrtszstk) [inline recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb:5]: should it also be in separate PR?

File: recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb [hunk 29]
```diff
@@ -35,7 +35,7 @@ SRCREV_libwebrtc = "a6522062f83cac380a4544e036114015cf061ffa"
 S = "${WORKDIR}/src"
 B = "${WORKDIR}/src/out/Linux-${GN_TARGET_ARCH_NAME}"
 
-inherit gn-fetcher features_check pkgconfig
+inherit gn-fetcher pkgconfig
```
Comment (xbrtszstk) [inline recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb:38]: it can be as separate PR as well, because as I guess feature_check is not used at all

File: recipes-graphics/toyota/flutter-auto_2.0.bb (hunk 25 above)
Comment (xbrtszstk) [reply recipes-graphics/toyota/flutter-auto_2.0.bb:30]: unfortunately it's necessary. in other case this package is unbuildable, because of wrong location of LICENSE and CMakeLists.txt files

Comment (jwinarske) [reply recipes-graphics/toyota/flutter-auto_2.0.bb:30]: I think there need to be some cherry picks to master from scarthgap first

File: recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb (hunk 28 above)
Comment (jwinarske) [reply recipes-graphics/libwebrtc/libwebrtc_125.6422.06.bb:5]: this would fall under agnostic PR, so not related to required master changes

File: recipes-graphics/toyota/flutter-auto_2.0.bb (hunk 25 above)
Comment (xbrtszstk) [reply recipes-graphics/toyota/flutter-auto_2.0.bb:30]: you might not have noticed, but I started from scarthgap

**Comment by jwinarske** [comment]
@xbrtszstk These are the steps for master
Step 1 - Rebase master to scarthgap.  Acquire all the commits from scarthgap since master was created
Step 2 - minimum changes required to build master; nothing agnostic
Step 3 - CI jobs
Step 4 - <optional> minimum agnostic PR; changes which can be cherry picked to scarthgap, kirkstone, and dunfell.  Not related to master required changes.
